lib/standalone.js

```diff
@@ -290,6 +300,27 @@ function getFormatterFunction(selected) {
 	return formatters.json;
 }
 
+/**
+ * @param {InternalApi} stylelint
+ * @param {string[]} fileList
+ * @param {string[]} filePaths
+ */
+async function filterByExtensions(stylelint, fileList, filePaths) {
+	const { cwd, extensions, globbyOptions } = stylelint._options;
+	const globCWD = globbyOptions?.cwd || cwd;
+	const directories = fileList
+		.filter((p) => fs.existsSync(p) && fs.lstatSync(p).isDirectory())
+		.map((p) => path.relative(globCWD, p));
+	const extList = extensions?.length ? extensions : DEFAULT_EXTENSIONS;
+
+	return filePaths.filter((p) => {
+		if (directories.some((directory) => p.startsWith(directory)))
+			return extList.some((extension) => p.endsWith(`.${extension}`));
```

Comment:
[suggest] We can rename local vars readable and short enough like this:

```suggestion
	return filePaths.filter((file) => {
		if (directories.some((dir) => file.startsWith(dir))) {
			return extList.some((ext) => p.endsWith(`.${ext}`));
		}
```

Reply:
You made a typo but I agree with you.